src/SConscript

```diff
@@ -662,43 +670,229 @@ def BuildRelease(target, source, env):
-            nsis_path = 'makensis'
-            command_flag = '-'
+            # TODO(XXX) How to handle that ? what does this exactly means ?
+            raise Exception('Unimplemented...')
 
-        if not nsis_path:
-            print "Cannot find NSIS. Do you have it installed?"
+        if not wix_directory:
+            raise Exception ('Cannot find WIX Toolkit. Do you have it installed?')
         else:
-            # Call the NSIS build
-            buildwin64 = "/Dx64=1" if build.machine_is_64bit else ''
-            staticdeps = "/DSTATICDEPS=1" if build.static_dependencies else ''
-            command = '%(path)s /DPACKAGE_NAME=\"%(package_name)s\" /DPRODUCT_VERSION=%(version)s /DQTDIR=%(qtpath)s /DWINLIB_PATH=%(winlibpath)s %(64bit)s %(staticdeps)s build\\nsis\\Mixxx.nsi' % \
-                {'path': nsis_path,
-                 'package_name': exe_name,
-                 'version': mixxx_version,
-                 'qtpath': build.env['QTDIR'],
+            print "    Found Wix Toolset in " + wix_path
+
+        WinSDK_path = 'build\\wix'
+
+        if not os.path.isfile(WinSDK_path + '\wisubstg.vbs'):
+            raise Exception ('can not find ' + WinSDK_path + '\wisubstg.vbs')
+
+        if not os.path.isfile(WinSDK_path + '\WiLangId.vbs'):
+            raise Exception ('can not find ' + WinSDK_path + '\WiLangId.vbs')
+
+        # The default language
+        defaultLanguage="en-us"
+        # The langIds contained in the installer. starting with LangId of the default language
+        langIds="1033"
+
+        winArch = "x64" if build.machine_is_64bit else "x86"
+
+        # Auto-create wxs file for each subdir and compile them
+        print "*** Building intermediate files"
+        for subdir in next(os.walk(dist_dir))[1]:
+            print "    " + dist_dir + "\\" + subdir
+            # Exclude doc, promo tracks and imageformats helper DLLs, they are bundled elsewhere
+            if subdir in ["doc", "promo", 'imageformats']:
+                continue
+            command = '"%(wix)s\\heat.exe" dir %(distdir)s\%(sub)s -nologo -sfrag -suid -ag -srd -cg %(sub)sComp -dr %(sub)sDir -out build\wix\subdirs\%(sub)s.wxs -sw5150 -var var.%(sub)sVar' % \
+                {'wix': wix_path,
+                 'distdir': dist_dir,
+                 'sub': subdir}
+            print "Using Command: " + command
+            subprocess.check_call(command)
+            command = '"%(wix)s\\candle.exe" -nologo -dWINLIBPATH=%(winlibpath)s -dPlatform=%(arch)s -d%(sub)sVar=%(distdir)s\%(sub)s -arch %(arch)s -out build\wix\subdirs\%(sub)s.wixobj build\wix\subdirs\%(sub)s.wxs' % \
+                {'wix': wix_path,
+                 'winlibpath': mixxx_winlib_path,
+                 'arch': winArch,
+                 'distdir': dist_dir,
+                 'sub': subdir}
+            print "Using Command: " + command
+            subprocess.check_call(command)
+
+        # Handle promo tracks if any
```

Comment:
I would drop promo track stuff entirely. It's ages old and the code for it is gone.